lib/compress.js

```diff
@@ -338,7 +338,7 @@ merge(Compressor.prototype, {
                     if (ref.scope.uses_eval || ref.scope.uses_with) break;
 
                     // Constant single use vars can be replaced in any scope.
-                    if (var_decl.value.is_constant(compressor)) {
+                    if (!(var_decl.value instanceof AST_RegExp) && var_decl.value.is_constant(compressor)) {
```

Comment:
I would say that you put the exception in those locations then, instead of this one, since it's clearly not a constant.